Validation/EcalDigis/plugins/EcalSelectiveReadoutValidation.h

```diff
@@ -789,13 +789,15 @@ class EcalSelectiveReadoutValidation : public DQMOneEDAnalyzer<> {
   /** Energy deposited in ECAL barrel crystals. Eta index starts from 0 at
    * eta minimum and phi index starts at phi=0+ in CMS std coordinate system.
    */
-  energiesEb_t ebEnergies[nEbEta][nEbPhi];
+  std::vector<std::vector<energiesEb_t>> ebEnergies =
+      std::vector<std::vector<energiesEb_t>>(nEbEta, std::vector<energiesEb_t>(nEbPhi));
```

Comment:
Why is std::vector better than C array? Why not std::array?

Reply:
The `energiesEb_t` `energiesEe_t` were very large arrays together accounting to about 4.4 MB, and were causing the issue as mentioned here: https://github.com/cms-sw/cmssw/issues/35004#issuecomment-905885016 
By using vectors the memory is allocated on the heap, instead of on the stack as when using arrays, and so the issue with memory running out is resolved.

Reply:
Just to clarify that the EDModules are allocated on the heap. The "problem" (I believe) is that the size of this class is so large that the UBSAN tool thinks an object of this class are accessed through invalid pointer when the base class methods are called. Moving these arrays out of the objects direct storage would work around of the "false positive" (that is similar as moving memory allocations from stack to heap).

Something like `std::unique_ptr<std::array<std::array<energiesEb_t, nEbPhi>, nEbEta>>` should work as well.